redpanda/resources/resourcegroup/data_resourcegroup.go

```diff
@@ -54,11 +54,13 @@ func datasourceResourceGroupSchema() schema.Schema {
 	return schema.Schema{
 		Attributes: map[string]schema.Attribute{
 			"id": schema.StringAttribute{
-				Required:    true,
+				Computed:    true,
```

Comment:
Not sure about computed here

https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-behaviors

Computed will hide this value in the plan stage. If we're expecting the user to sometime set it that may not be correct even if we can set it from the API.

Reply:
`aws_eip` does it this same way. It's similar in that it supports looking up an EIP by either `id` or `public_ip`, and also returning those same values on the object: https://github.com/hashicorp/terraform-provider-aws/blob/50a8f2c94bf304a43b2a950a81b7473e3e45e18d/internal/service/ec2/ec2_eip_data_source.go#L61

Reply:
I also think this is correct that it's _not_ available at the plan stage — the user might put in an ID or name that doesn't match any existing resource group. It needs to actually run to figure out if there's a resource group that can be used by downstream resources.